wagtailpolls/views/vote.py

```diff
@@ -33,7 +33,7 @@ def _vote(request, poll_pk):
     try:
         int(poll_pk)
     except ValueError:
-        return HttpResponse("<h1>{0}</h1>".format(_('Oops, there is no poll to vote on!')), status=500)
+        return Http404("<h1>{0}</h1>".format('Oops, there is no poll to vote on!'))
```

Comment:
`raise Http404(...)`  or `return HttpResponseNotFound(...)`, but not `return Http404(...)`. [See the Django docs for more information](https://docs.djangoproject.com/en/1.10/topics/http/views/#the-http404-exception)